specs/DisableCrashReporting.md

```diff
@@ -2,47 +2,48 @@ Process Info
 ===
 
 # Background
-Currently, when a WebView2 is created a WebView2 crashpad process will be created in addition to starting a WebView2 browser process. If any WebView2 process crashes, the crashpad process will create crash dumps and send them to Microsoft for diagnosis. This document covers new APIs to allow the end developer to customize crash reporting. They can set the `CoreWebView2EnvironmentOptions.IsCustomCrashReportingEnabled` property to not have crash dumps sent to Microsoft and use the `CoreWebView2Environment.CrashDumpFolderPath` property to locate crash dumps and do whatever they like with them instead.
+Currently, when a WebView2 is created a WebView2 crashpad process will be created in addition to starting a WebView2 browser process. If any WebView2 process crashes, the crashpad process will create crash dumps and send them to Microsoft for diagnosis. This document covers new APIs to allow the end developer to customize crash reporting. They can set the `CoreWebView2EnvironmentOptions.IsCustomCrashReportingEnabled` property to not have crash dumps sent to Microsoft and use the `CoreWebView2Environment.CrashDumpFolderPath` property to locate crash dumps and do customization with them instead.
 
 # Examples
 ## WinRT and .NET   
 ```c#
 
 /// Create WebView Environment with option
-auto options = new CoreWebView2EnvironmentOptions();
-options.DisableCrashReporting = true;
-auto environment = await CoreWebView2Environment.CreateAsync(BrowserExecutableFolder, UserDataFolder, options);
+void CreateEnvrionmentWithOption()
+{
+    CoreWebView2EnvironmentOptions options = new CoreWebView2EnvironmentOptions();
+    options.IsCustomCrashReportingEnabled = true;
+    CoreWebView2Environment environment = await CoreWebView2Environment.CreateAsync(BrowserExecutableFolder, UserDataFolder, options);
+}
 
-/// Webview creation and setup of completion handler are out of scope of sample
-void WebView_CoreWebView2InitializationCompleted(object sender, CoreWebView2InitializationCompletedEventArgs e)
+void DirFileChanged()
 {
-    if (e.IsSuccess)
-    {
-        // Get the crash dump folder
-        String crashDumpFolder = webView.CoreWebView2.Environment.CrashDumpFolder();
-    }
+    String crashDumpFolder = webView.CoreWebView2.Environment.CrashDumpFolderPath;
+    ProcessNewCrashDumps(crashDumpFolder);
 }
 
 ```
 ## Win32 C++
 ```cpp
-void AppWindow::ToggleCrashReportingUploadConsent()
+void AppWindow::InitializeWebView()
 {
-    // Disable crash report
     auto options = Microsoft::WRL::Make<CoreWebView2EnvironmentOptions>();
-    CHECK_FAILURE(options->put_DisableCrashReporting(TRUE));
+    CHECK_FAILURE(options->put_IsCustomCrashReportingEnabled(TRUE));
+    // ... other option properties
+
+    // ... CreateCoreWebView2EnvironmentWithOptions
 }
 
-void AppWindow::GetCrashDumpFolder()
+void AppWindow::DirFileChanged()
```

Comment:
In both of these samples can we show how to hook up DirFileChanged? How does that method get called?

Reply:
Here is some samples https://docs.microsoft.com/en-us/windows/win32/fileio/obtaining-directory-change-notifications. But do we want to write the whole thing in our sample code and sample app?

Reply:
Agreed here, we should show an example of how to use the DirFileChanged API. Can we do what Dave recommended in the comment above? https://github.com/MicrosoftEdge/WebView2Feedback/pull/2769/files#r965318889

Reply:
DirFileChanged is not an WebView API. I am still verifying on when dump file are written to disk

Reply:
Sorry, I meant using the API to get the crash dump folder.